pulp_2to3_migration/app/migration.py

```diff
@@ -320,6 +324,89 @@ def create_repo_version(progress_rv, pulp2_repo, pulp3_remote=None):
+
+        NOTE: The logic is copied from pulpcore.app.files.validate_file_paths().
+
+        This function returns the first dupe or overlap it detects. We use a trie (or
+        prefix tree) to keep track of which paths we've already seen.
+
+        Args:
+            paths (iterable of str): An iterable of strings each representing a relative path
+
+        Returns:
+            str: a path which overlaps or duplicates another
+
+        """
+        path_trie = StringTrie(separator="/")
+        for path in paths:
+            if path in path_trie:
+                # path duplicates a path already in the trie
+                return path
+
+            if path_trie.has_subtrie(path):
+                # overlap where path is 'a/b' and trie has 'a/b/c'
+                return path
+
+            prefixes = list(path_trie.prefixes(path))
+            if prefixes:
+                # overlap where path is 'a/b/c' and trie has 'a/b'
+                return path
+
+            # if there are no overlaps, add it to our trie and continue
+            path_trie[path] = True
+
+    def resolve_path_overlap(version):
+        """
+        Remove content which path overlaps some other.
+
+        If it's a duplicated path, remove the older content.
+
+        If something is absolutely wrong and we were not able to resolve conflicts,
+        repo version creation will fail later.
+
+        Paths can be overlapping because of an old Pulp 2 bug.
+
+        Args:
+            version(pulpcore.app.model.RepositoryVersion): incomplete version which needs path
+                overlap resolution
+
+        """
+        paths = ContentArtifact.objects.filter(content__pk__in=version.content).values_list(
+            "relative_path", flat=True
+        )
+        paths = list(paths)
+        max_conflicts = version.content.count() - 1
+        for i in range(max_conflicts):
+            bad_path = detect_path_overlap(paths)
+            if not bad_path:
+                # no path overlaps, we are good
+                break
+
+            # Content Artifacts with conflicting relative paths ordered by pulp2 creation time
+            cas_with_conflicts = ContentArtifact.objects.filter(
+                content__pk__in=version.content, relative_path=bad_path
+            ).order_by('-content__pulp2content__pulp2_last_updated')
+
+            conflict_count = cas_with_conflicts.count()
+            if conflict_count > 1:
+                # There are duplicated paths and we need to keep the newest content in the version.
+                # The query result is ordered desc by time so we remove all but the first content.
+                content_to_remove = [ca.content.pk for ca in cas_with_conflicts[1:]]
+                version.remove_content(Content.objects.filter(pk__in=content_to_remove))
+                # exclude the duplicated paths from further search
+                for j in range(conflict_count - 1):
```

Comment:
Can we get an INFO log here, "Removing {conflict_count -1} instance of {bad_path}" here, so we have an answer when someone says "this repo had N rpms in Pulp2, and only N-M in pulp3 - what happened?!?"

Reply:
my suggestion would be to not pollute logging with N messages, instead how about notifying the user once about the fact that there is content that has some path overlap issue. I doubt user needs to know that path exactly is being solved.
```
for i in range(max_conflicts):
    ...
else:
   log.info('Some content had  overlapping paths and was removed')
``

Thoughts?

Reply:
Or maybe not. It seems like The for loop will will always hit 'break' sooner or later.

Reply:
we can always compare the len of paths before entering the loop and after exiting it;)

Reply:
Yeah, this code executes only when there are conflicts. For the majority of repo verisons, nothing will be printed.